lib/mongo/server/connection.rb

```diff
@@ -167,6 +167,7 @@ def ping
       def timeout
         @timeout ||= options[:socket_timeout]
       end
+      alias :socket_timeout :timeout
```

Comment:
Even though this is all private api, in general, I'd rather not remove or rename methods unless I'm planning to do a major release. And in that case, I'd mark it as deprecated leading up to the release.
Ideally, I would just rename it but I need to stick with semantic versioning rules.